src/cpp/shared/combat/army.cpp

```diff
@@ -2279,11 +2278,10 @@ signed int army::SetSpellInfluence(int effectType, signed int strength) {
       this->frameInfo.stepTime = (signed __int64)((double)this->frameInfo.stepTime * 0.65);
       break;
     case EFFECT_SLOW:
-      this->CancelIndividualSpell(0);
-      v4 = (signed int)this->creature.speed + 1;
-      this->creature.speed = ((signed int)v4 - v4) >> 1;
-      if (this->creature.creature_flags & 2)
-        this->creature.creature_flags -= 2;
+      this->CancelIndividualSpell(EFFECT_HASTE);
+      this->creature.speed = (this->creature.speed + 1) / 2;
```

Comment:
Wow! I had always thought that Slow was a flat -2 to speed. Just tested in the original game; it's not.